### ISSUE TITLE
Dependency check does not support "replace" keyword in go.mod file

### ISSUE THREAD

**Issue by azharem** [dependency-check/DependencyCheck#2904]
**Describe the bug**
We have multiple golang modules in our repo for various microservices. Each module has its own go.mod file.  When we run dependency check, the modules with go.mod files that do not have "replace" command pass but the ones with "replace" command fail.

**Version of dependency-check used**
The problem occurs using version 6.0.2 of the cli.

**Log file**
```
[WARN] An error occurred while analyzing '/builds/xxxx/bbb/src/wwww/nnnn/go.mod' (Golang Mod Analyzer).
[INFO] Finished Golang Mod Analyzer (8 seconds)
[INFO] Finished Dependency Merging Analyzer (0 seconds)
[INFO] Finished Version Filter Analyzer (0 seconds)
[INFO] Finished Hint Analyzer (0 seconds)
[INFO] Created CPE Index (1 seconds)
[INFO] Finished NPM CPE Analyzer (2 seconds)
[INFO] Created CPE Index (2 seconds)
[INFO] Finished CPE Analyzer (11 seconds)
[INFO] Finished False Positive Analyzer (0 seconds)
[INFO] Finished NVD CVE Analyzer (4 seconds)
[INFO] Finished Node Audit Analyzer (2 seconds)
[INFO] Finished RetireJS Analyzer (0 seconds)
[INFO] Finished Sonatype OSS Index Analyzer (4 seconds)
[INFO] Finished Vulnerability Suppression Analyzer (0 seconds)
[INFO] Finished Dependency Bundling Analyzer (0 seconds)
[INFO] Analysis Complete (35 seconds)
[ERROR] Error parsing stream
[ERROR] Error parsing stream
```

**To Reproduce**
1. Create two modules with two different go.mod files
2. Use "replace" in only one of the go.mod files
3. Scan both files
4. The one with "replace" will fail.

**Sample go.mod file that failed**
```text
module gitlab/zzz/aaa

go 1.15

require (
    github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
    github.com/crewjam/httperr v0.2.0 // indirect
    github.com/crewjam/saml v0.4.1
    github.com/go-chi/chi v4.1.2+incompatible
    github.com/russellhaering/goxmldsig v0.0.0-20200902171629-2e1fbc2c5593 // indirect
    github.com/stretchr/testify v1.6.1
    github.com/swaggo/swag v1.6.7
    gitlab/zzz/bbbb v0.0.3
    golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a // indirect
    gopkg.in/yaml.v2 v2.3.0 // indirect
)

replace gitlab/zzz/bbbb => ../bbbb
```

**Comment by jeremylong** [comment]
What do you get when you run

```
go list -json -m all
```

Also, please create a log file by including `dependency-check.sh --log odc.log ...` to the execution.

**Comment by fpytloun** [comment]
We are facing same issue with version 6.0.3 even that there's no `replace` in go.mod. This makes depcheck unusable for most of our projects.

Running with
```
dependency-check --enableExperimental --noupdate --scan go.mod -f ALL --log odc.log
```

Attaching log and output of `go list`:

[golist.txt](https://github.com/jeremylong/DependencyCheck/files/5657848/golist.txt)
[odc.log](https://github.com/jeremylong/DependencyCheck/files/5657849/odc.log)

**Comment by jeremylong** [comment]
A fix for this will be included in 6.0.4 - see #2987.
